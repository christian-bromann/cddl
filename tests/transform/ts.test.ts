import url from 'node:url'
import path from 'node:path'
import { describe, it, expect } from 'vitest'

import CDDL, { ParseTargets } from '../../src/index.js'

const __dirname = url.fileURLToPath(new URL('.', import.meta.url))
const webdriverCDDL = path.join(__dirname, '..', '__fixtures__', 'webdriver.cddl')

describe('transform TS', () => {
    it('should return valid TS types', () => {
        const ts = CDDL.parse(webdriverCDDL, { target: ParseTargets.TS })
        expect(ts).toMatchSnapshot()
    })
})