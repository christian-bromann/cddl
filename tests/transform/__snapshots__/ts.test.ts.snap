// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`transform TS > should return valid TS types 1`] = `
"type Message = CommandResponse | ErrorResponse | Event;

interface ErrorResponse extends Extensible {
  id?: JsUint | null,
  error: ErrorCode,
  message: string,
  stacktrace?: string,
}

type ResultData = BrowsingContextResult | EmptyResult | NetworkResult | ScriptResult | SessionResult;
type ErrorCode = \\"invalid argument\\" | \\"invalid session id\\" | \\"no such alert\\" | \\"no such frame\\" | \\"no such handle\\" | \\"no such node\\" | \\"no such script\\" | \\"session not created\\" | \\"unknown command\\" | \\"unknown error\\" | \\"unsupported operation\\";
type JsInt = number;
type JsUint = number;
type Handle = string;
type SharedId = string;
interface UndefinedValue { type: \\"undefined\\" }
interface NullValue { type: null }

interface StringValue {
  type: \\"string\\",
  value: string,
}

type SpecialNumber = \\"NaN\\" | \\"-0\\" | \\"Infinity\\" | \\"-Infinity\\";

interface NumberValue {
  type: \\"number\\",
  value: number | SpecialNumber,
}

interface SessionCapabilitiesRequest {
  alwaysMatch?: SessionCapabilityRequest,
  firstMatch?: SessionCapabilityRequest[],
}"
`;
